tools/render-test/shader-input-layout.cpp

```diff
@@ -1100,7 +1126,109 @@ namespace renderer_test
         return SLANG_OK;
     }
 
+    void loadDataIntoHalf(uint16_t& out, const uint8_t& in)
+    {
+        out = FloatToHalf((float(in) / 255.0f));
+    }
+    void loadDataIntoFloat(float& out, const uint8_t& in)
+    {
+        out = (float(in)/255.0f);
+    }
+    template<typename T>
+    void loadDataIntoUint(T& out, const uint8_t& in)
+    {
+        out = T(in);
+    }
+    template<typename T>
+    void loadDataIntoInt(T& out, const uint8_t& in)
+    {
+        // well define int8_t = uint8_t
+        if constexpr (std::is_same<int8_t, T>())
+        {
+            if (in > 127)
+            {
+                out = T(127);
+                return;
+            }
+        }
```

Comment:
Why is only one type treated specially?

Reply:
uint8_t is the src type, max size is 255.

int16, uint16, half all can safely store the type lower and upper bounds in normal c++ rules with an explicit cast or with logic (FloatToHalf).

int8_t is the only type which cannot cast 255 into itself within 0-128, the spec states that bit pattern is preserved, so this would end-up being [0,127],[-1,-128]. Since testing code does not assign negatives I decided to follow this pattern and just keep things consistent with a positive only int.

Reply:
I can add a way to support negative fairly trivially if you prefer and believe it is the correct method